src/Types.jl

```diff
@@ -236,6 +237,13 @@ end
 Base.:(==)(t1::Project, t2::Project) = all(x -> (getfield(t1, x) == getfield(t2, x))::Bool, fieldnames(Project))
 Base.hash(t::Project, h::UInt) = foldr(hash, [getfield(t, x) for x in fieldnames(Project)], init=h)
 
+# only hash the deps and compat fields as they are the only fields that affect a resolve
+function project_resolve_hash(t::Project, h::UInt = UInt(0))
+    iob = IOBuffer()
+    println(iob, t.deps)
```

Comment:
This relies on printing of this dict (which also isn't ordered). Perhaps something like:
```suggestion
    foreach(((name, uuid),) -> println(io, name, "=", uuid), sort!(collect(t.deps); by=first))
```